src/app/services/training/service.py

```diff
@@ -32,11 +22,6 @@ class TrainingService(BaseModel):
 
     model_config = ConfigDict(arbitrary_types_allowed=True)
 
-    filter_config: FilterConfig = FilterConfig()
-    mapping_config: MappingConfig = MappingConfig()
-    feature_engineer_config: FeatureEngineerConfig = FeatureEngineerConfig()
-    selection_config: SelectionConfig = SelectionConfig()
-
     @staticmethod
```

Comment:
for this case:

```suggestion
    memory = Memory(
                location=str(Settings.MODEL_DIRECTORY / "pipeline_cache"), verbose=0
            )
```
Maybe you should consider to create the directory if it isn't exist.

```suggestion
    // ...existing code...
            cache_dir = Settings.MODEL_DIRECTORY / "pipeline_cache"
            cache_dir.mkdir(parents=True, exist_ok=True)

            memory = Memory(location=str(cache_dir), verbose=0)
    // ...existing code...
```

Reply:
Directory is created automatically when called so there is no need to ensure the directory